crates/primitives/src/prune/mod.rs

```diff
@@ -6,4 +6,43 @@ mod target;
 pub use checkpoint::PruneCheckpoint;
 pub use mode::PruneMode;
 pub use part::{PrunePart, PrunePartError};
-pub use target::PruneModes;
+pub use target::{PruneModes, MINIMUM_PRUNING_DISTANCE};
+
+use crate::{Address, BlockNumber};
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+/// Configuration for pruning receipts not associated with logs emitted by the specified contracts.
+#[derive(Debug, Clone, PartialEq, Eq, Default, Serialize, Deserialize)]
+pub struct ContractLogsPruneConfig(pub Vec<(PruneMode, Address)>);
```

Comment:
should this be a btreemap<address, PruneMode> instead?

Reply:
took a stylistic choice on how its represented on the `toml` file, but I'm open to change:

btreemap:
```
[parts.contract_logs_filter.0x00000000219ab540356cbb839cbe05303d7705fa]
before = 11052984

[parts.contract_logs_filter.0xf08a50178dfcde18524640ea6618a1f965821715]
before = 3802888
```

vec:
```
contract_logs_filter = [
    [
        { before = 11052984 },
        "0x00000000219ab540356cbb839cbe05303d7705fa",
    ],
    [
        { before = 3802888 },
        "0xf08a50178dfcde18524640ea6618a1f965821715",
    ],
]
```

Reply:
@shekhirin

Reply:
I'd rather go with btreemap, vec looks like a json inside toml for me